File: packages/java-sdk/src/main/java/dev/blindfold/sdk/models/DetectedEntity.java
package dev.blindfold.sdk.models;

import com.google.gson.annotations.SerializedName;

public class DetectedEntity {
    private String type;
    private String text;
    private int start;
    private int end;
    private double score;

    // Default constructor for Gson
    public DetectedEntity() {}

    public DetectedEntity(String type, String text, int start, int end, double score) {
        this.type = type;
        this.text = text;
        this.start = start;
        this.end = end;
        this.score = score;
    }

    public String getType() { return type; }
    public String getText() { return text; }
    public int getStart() { return start; }
    public int getEnd() { return end; }
    public double getScore() { return score; }

    @Override
    public String toString() {
        return "DetectedEntity{type='" + type + "', text='" + text + "', start=" + start + ", end=" + end + ", score=" + score + "}";
    }
}

File: packages/dotnet-sdk/src/Blindfold.Sdk/Models/DetectedEntity.cs
using System.Text.Json.Serialization;

namespace Blindfold.Sdk.Models;

public class DetectedEntity
{
    [JsonPropertyName("type")]
    public string Type { get; set; } = "";

    [JsonPropertyName("text")]
    public string Text { get; set; } = "";

    [JsonPropertyName("start")]
    public int Start { get; set; }

    [JsonPropertyName("end")]
    public int End { get; set; }

    [JsonPropertyName("score")]
    public double Score { get; set; }
}
